app/models/concerns/maintenance_task_instrumentation.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module MaintenanceTaskInstrumentation
+  extend ActiveSupport::Concern
```

Comment:
Does this make sense to keep as a concern?  What if it were just a singleton?  Something where we could just call:

```
MaintenanceTaskInstrumentation.instrument_as_maintenance_task(...) { ... }
```

This doesn't need access to the calling class right?

Reply:
Sure, we can use a singleton. I had originally hoped to use around_perform to minimize code churn but that doesn't seem to work; probably due to interaction with the existing lifecycle hooks in BaseJob.

Maybe `instrument_as_maintenance_task` should just be a method on BaseJob if we need this mechanism on many jobs